packages/hash/api/src/model/entityType.model.ts

```diff
@@ -206,9 +256,14 @@ class __EntityType {
   static async getEntityTypeBySchema$id(
     client: DBClient,
     params: { schema$id: string },
-  ): Promise<EntityType | null> {
+  ): Promise<EntityType> {
     const dbEntityType = await client.getEntityTypeBySchema$id(params);
-    return dbEntityType ? new EntityType(dbEntityType) : null;
+    if (!dbEntityType) {
+      throw new Error(
+        `Critical: Could not find EntityType by Schema$id = ${params.schema$id}`,
+      );
+    }
+    return new EntityType(dbEntityType);
   }
```

Comment:
This was done to consolidate the error. But you're right, the error should be thrown only when it's critical. I'll add this error check where it's needed instead of here